### PR TITLE
Set up the python infrastructure

### PR DESCRIPTION
**Pull request by natemcintosh**
Just the basics for now, including updating the pre-commit setup.

### CODE REVIEW

**Comment by natemcintosh** [comment]
Oh, hold on, forgot to put in basic python CI

File: .github/workflows/python.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        python-version-file: ".python-version"
+
+    - name: Install the project
+      run: uv sync --all-extras --dev
+
+    - name: Run tests
+      # For example, using `pytest`
+      run: uv run pytest
+
+    - name: Minimize uv cache
```
Comment (amondal2) [inline .github/workflows/python.yaml:33]: What does this do? Do caches persist across CI runs? Just curious

**Review by amondal2 — APPROVED** [review]
Nice; excited to try out `uv`

File: .github/workflows/python.yaml (hunk 1 above)
Comment (natemcintosh) [reply .github/workflows/python.yaml:33]: Each [repo gets](https://github.com/CDCgov/cfa-rt-postprocessing/actions/caches) 10GB of cache space, which is honestly a huge amount. This just helps keep things a little smaller, while still giving most of the benefits of a full cache.

**Pull request merged**